### ISSUE TITLE
Add attachments to user feedback

### ISSUE THREAD

**Issue by franNadalesDN** [getsentry/sentry#49946]
### Problem Statement

I want users to be able to report issues and feedback with attachments such log files or images.

### Solution Brainstorm

Use similar structure as events for User feedback.

### Product Area

User Feedback

**Comment by bruno-garcia** [comment]
Today the feedback you can submit with the SDK takes an `event_id`. So you need to capture an exception or at least a message to attach the feedback to. The same way you can use that `event_id` to tie attachments to it.

I believe we can improve this by letting you capture feedbacks without the need to capture an event (error/msg), and we recently did that to power [our new Widget](https://docs.sentry.io/product/user-feedback/#user-feedback-widget). So it's something we can look into going forward but for now I suggest using the captureMessage with Attachment + captureUserFeedback route

**Comment by billyvg** [comment]
We are also adding screenshotting to our User Feedback widget: https://github.com/getsentry/sentry/issues/63749

**Comment by bruno-garcia** [comment]
The new API could take a local scope:

```javascript
let feedback = {name:"", message:""}
Sentry.captureUserFeedback(feedback, scope => scope.addAttachment(...));
```

**Comment by ryan953** [comment]
The new/updated api to call is being implemented in https://github.com/getsentry/sentry-javascript/pull/11428

It'll be available on v8.0.0 of the SDK as `Sentry.captureFeedback()` and will very closely align with the other `captureException()` and `captureMessage()` methods we have.

The specific [method is implemented with this signature & types](https://github.com/getsentry/sentry-javascript/pull/11428/files#diff-21cd67620d89f67e1f63bbfc45f135d6ecbc2f11cb62477a50117c92d3c7aefe):
```
interface SendFeedbackParams {
  message: string;
  name?: string;
  email?: string;
  url?: string;
  source?: string;
  associatedEventId?: string;
}
interface EventHint {
  event_id?: string;
  captureContext?: CaptureContext;
  mechanism?: Partial<Mechanism>;
  syntheticException?: Error | null;
  originalException?: unknown;
  attachments?: Attachment[];
  data?: any;
  integrations?: string[];
}

function captureFeedback(
  feedbackParams: SendFeedbackParams,
  hint: EventHint & { includeReplay?: boolean } = {},
): string
```

Look out for it once #11428 lands, which should be part of 8.0.0-beta.6 and beyond.
